doc/tools/docgram/README.md

```diff
@@ -0,0 +1,194 @@
+Grammar extraction tool for documentation
+=========================================
+
+doc_grammar extracts Coq's grammar from .mlg files, edits it and inserts it in
+chunks into .rst files.  The current prototype generates Sphinx
+`productionlist` displays.  For `tacn` and `cmd`, the tool verifies that the
+.rst matches the grammar.  Updates to tactics and commands must be made
+manually since the grammar doesn't have names for them as it does for
+nonterminals.  `tacv` and `cmdv` are not processed at all.
+
+The mlg grammars present several challenges to generating an accurate grammar
+for documentation purposes:
+* The 30+ mlg files don't define an overall order in which nonterminals should
+appear in a complete grammar.
+* Even within a single mlg file, grammar rules and productions are often given
+in an order that's much different from what a reader of the documentation would
+expect.  In a small number of cases, the order used is necessary for technical
+reasons.
+* For a few nonterminals, some productions are only available if certain plugins
+are activated (e.g. SSR).  Readers should be informed about these.
+* Some limited parts of the grammar are defined in OCaml, including lookahead symbols
+like `test_bracket_ident` and references to nonterminals in other files using
+qualified names such as `Prim.ident`.  A few symbols are defined multiple times,
+such as `scope` and `orient`.
+
+What the tool does
+------------------
+1. The tool reads all the mlg files and generates `fullGrammar`, which includes
+all the grammar without the actions for each production or the OCaml code.  This
+file is provided as a convenience to make it easier to examine the (mostly)
+unprocessed grammar of the mlg files with less clutter.  Nonterminals that use
+levels ("5" RIGHTA below) are modified, for example:
+
+```
+  tactic_expr:
+    [ "5" RIGHTA
+      [ te = binder_tactic -> { te } ]
+```
+
+becomes
+
+```
+tactic_expr5: [
+| binder_tactic
+| tactic_expr4
+]
+```
+
+2. The tool applies grammar editing operations specified by `common.edit_mlg` to
+generate `editedGrammar`.
+
+3. The tool uses `orderedGrammar` to give the desired order for the nonterminals
+and productions in the documented grammar.  `orderedGrammar` has
+the same format as `editedGrammar`.  The tool verifies that `orderedGrammar` is
+a permutation of `editedGrammar`.  Developers are responsible for editing
+`orderedGrammar`.
+
+4. The tool applies further edits to the grammar specified by `prodlist.edit_mlg`,
```

Comment:
Can you rename this file in `productionlist.edit_mlg` so that it is even clearer that it is the `edit_mlg` file for `productionlist`s?